libs/providers/vmware.py

```
@@ -1051,27 +1046,29 @@ def _resolve_datastore(disk_datastore_id: str | None) -> vim.Datastore:
-            elif disk_datastore_id == "non_xcopy_datastore_id":
-                if non_xcopy_datastore is None:
-                    raise VmCloneError(
-                        "Disk requested non-XCOPY datastore but copyoffload.non_xcopy_datastore_id is not configured"
-                    )
-                resolved = non_xcopy_datastore
-            elif disk_datastore_id is not None:
+            if disk_datastore_id is None:
+                resolved = target_datastore
+            else:
                 if not disk_datastore_id:
                     raise VmCloneError("Disk datastore_id is empty. Provide a valid MoID or omit the field.")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)